deployment/promotion/promote.py

```diff
@@ -75,33 +79,49 @@ def auth_context(host, email, devkey, workspace):
 def post(auth, path, body):
     auth['headers']['Content-Type'] = 'application/json'
     body['workspaceName'] = auth['workspace']
-    res = requests.post("https://{}{}".format(auth["host"], path), headers=auth['headers'], json=body)
-    if res.ok:
-        return res.json()
-    raise Exception(res.text)
+    try:
+        res = requests.post("https://{}{}".format(auth["host"], path), headers=auth['headers'], json=body)
+        res.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        if len(e.response.text) > 0:
+            print(f"Error: {e.response.text}")
+        raise Exception(e)
```

Comment:
We could re-raise the original exception object rather than casting it to a base `Exception`, if you'd like.

```suggestion
        raise e
```